WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/handlers/StartupUIThread.java

```diff
@@ -20,29 +20,30 @@
  */
 public class StartupUIThread implements Runnable {
 
-	private final StartUpHandler startUpHandler;
-
 	/** The preferences. */
 	private Preferences preferences;
 
+	/** Whether user has cancelled first registration. */
+	private boolean userRegistrationCancelled = false;
+
 	/** Constructor. */
-	public StartupUIThread(StartUpHandler startUpHandler,
-			Preferences preferences) {
-		this.startUpHandler = startUpHandler;
+	public StartupUIThread(Preferences preferences) {
 		this.preferences = preferences;
 	}
 
 	@Override
 	public void run() {
 		// the execution strategy is:
-		// (1) check user registration (2) wait until 1 is complete, or 1
-		// can be skipped (3) show workspace registration
+		// (1) check user registration (2) wait until user+project registration
+		// is completed, or 1 can be skipped (3) if user is already registered,
+		// check for workspace registration
 		if (WatchDogUtils.isEmpty(preferences.getUserid())) {
 			displayUserRegistrationWizard();
 		}
 
-		if (!WatchDogUtils.isEmpty(preferences.getUserid())) {
-			// In case the user aborted the preference dialog with cancel,
+		if (!WatchDogUtils.isEmpty(preferences.getUserid())
```

Comment:
Okay, I'll try like this: 
1) First WD installation, user starts user+project registration
2) Finishes user registration part, cancel during project registration (he doesn't want to register current workspace)
3) Flag is set on true, to avoid WD asking user whether or not he would like WD to monitor this workspace (since he cancelled, he obviously doesn't want this).
4) If a user later changes his workspace, this flag would be false, so another workspace will be checked, with offering just project registration wizard.